tests/storage_test.go

```diff
@@ -321,29 +339,39 @@ var _ = Describe("Storage", func() {
 							virtClient,
 							vmiPod,
 							vmiPod.Spec.Containers[0].Name,
-							[]string{"find", hostDiskDir, "-name", diskName, "-size", "1G"},
+							[]string{"find", hostdisk.GetMountedHostDiskDir("host-disk"), "-name", diskName, "-size", "1G"},
 						)
-						Expect(strings.Contains(output, diskPath)).To(BeTrue())
-
-						err = virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})
 						Expect(err).ToNot(HaveOccurred())
-
-						tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+						Expect(output).To(ContainSubstring(hostdisk.GetMountedHostDiskPath("host-disk", diskPath)))
 					})
 
-					AfterEach(func() {
-						if nodeName != "" {
-							tests.RemoveHostDiskImage(diskPath, nodeName)
-						}
+					It("should start with multiple hostdisks in the same directory", func() {
+						By("Starting VirtualMachineInstance")
+						// do not choose a specific node to run the test
+						vmi := tests.NewRandomVMIWithHostDisk(diskPath, v1.HostDiskExistsOrCreate, "")
+						tests.AddHostDisk(vmi, filepath.Join(hostDiskDir, "another.img"), v1.HostDiskExistsOrCreate, "anotherdisk")
+						tests.RunVMIAndExpectLaunch(vmi, 30)
+
+						By("Checking if disk.img has been created")
+						vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+						nodeName = vmiPod.Spec.NodeName
+						output, err := tests.ExecuteCommandOnPod(
+							virtClient,
+							vmiPod,
+							vmiPod.Spec.Containers[0].Name,
+							[]string{"find", hostdisk.GetMountedHostDiskDir("anotherdisk"), "-name", "another.img", "-size", "1G"},
+						)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output).To(ContainSubstring(hostdisk.GetMountedHostDiskPath("anotherdisk", filepath.Join(hostDiskDir, "another.img"))))
```

Comment:
Shouldn't we do this check for both disks, just to make sure we didn't break the first one with this patch?

Other than that lgtm

Reply:
Done